lib/rules/no-unsafe.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @fileoverview Prevent usage of UNSAFE_ methods
+ * @author Sergei Startsev
+ */
+
+'use strict';
+
+const Components = require('../util/Components');
+const astUtil = require('../util/ast');
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent usage of UNSAFE_ methods',
+      category: 'Best Practices',
+      recommended: true,
+      url: docsUrl('no-unsafe')
+    },
+    schema: []
+  },
+
+  create: Components.detect((context, components, utils) => {
+    /**
+     * Returns a list of unsafe methods
+     * @returns {Array} A list of unsafe methods
+     */
+    function getUnsafeMethods() {
+      return [
+        'UNSAFE_componentWillMount',
+        'UNSAFE_componentWillReceiveProps',
+        'UNSAFE_componentWillUpdate'
+      ];
+    }
+
+    /**
+     * Checks if a passed method is unsafe
+     * @param {string} method Life cycle method
+     * @returns {boolean} Returns true for unsafe methods, otherwise returns false
+     */
+    function isUnsafe(method) {
+      const unsafeMethods = getUnsafeMethods();
+      return unsafeMethods.indexOf(method) !== -1;
+    }
+
+    /**
+     * Reports the error for an unsafe method
+     * @param {ASTNode} node The AST node being checked
+     * @param {string} method Life cycle method
+     */
+    function checkUnsafe(node, method) {
+      if (!isUnsafe(method)) {
+        return;
+      }
+
+      context.report({
+        node: node,
+        message: `Do not use ${method}`
```

Comment:
the message should explain why, and link to more info

Reply:
Added details